ZukoAzula2016/src/org/team1540/zukoazula/Shooter.java

```diff
@@ -12,40 +15,64 @@
 import ccre.cluck.Cluck;
 import ccre.ctrl.ExtendedMotor;
 import ccre.ctrl.ExtendedMotorFailureException;
+import ccre.drivers.ctre.talon.TalonExtendedMotor;
 import ccre.frc.FRC;
 import ccre.time.Time;
+import ccre.timers.ExpirationTimer;
+import ccre.timers.PauseTimer;
 
 public class Shooter {
-    private static final ExtendedMotor shooterLeftCAN = FRC.talonCAN(10);
-    private static final ExtendedMotor shooterRightCAN = FRC.talonCAN(11);
-    private static final FloatInput encoder = FRC.encoder(0, 1, false, FRC.startTele);
-
-    private static final ArbitratedBoolean shooterTrigger = ZukoAzula.behaviors.addBoolean();
+    private static final BehaviorArbitrator rollerArb = new BehaviorArbitrator("Rollers");
+    private static final ArbitratedFloat rollerSpeed = rollerArb.addFloat();
 
     public static void setup() throws ExtendedMotorFailureException {
-        BooleanInput trigger = ZukoAzula.controlBinding.addBoolean("Shoot");
-        shooterTrigger.attach(ZukoAzula.teleop, trigger);
-        shooterTrigger.attach(ZukoAzula.pit, trigger);
-        FloatOutput shooterMotors = shooterLeftCAN.simpleControl(FRC.MOTOR_FORWARD).combine(shooterRightCAN.simpleControl(FRC.MOTOR_REVERSE));
-        bangBangControl(encoder, shooterTrigger, ZukoAzula.mainTuning.getFloat("Shooter Velocity", 1), 0, 1).send(shooterMotors);
-        Cluck.publish("Shooter Motors", shooterMotors);
+        rollerSpeed.send(FRC.talonSimpleCAN(8, FRC.MOTOR_FORWARD));
+
+        BangBangTalon shooter = new BangBangTalon(makeLinkedTalons(), "Shooter");
+
+        // passively roll to the wheel in teleop, while the shooter is stopped.
+        rollerSpeed.attach(rollerArb.addBehavior("Passive Roll", FRC.inTeleopMode().and(shooter.isStopped)), ZukoAzula.mainTuning.getFloat("Roller Passive Speed", 0.1f));
+
+        PauseTimer actuallyFiring = new PauseTimer(500);
```

Comment:
This timer doesn't allow us to shoot directly after picking up a ball, keeping the shooter running at all times.Do we need to have this timer?